pkg/models/gacha.go

```diff
@@ -22,42 +69,87 @@ func init() {
 	db.AutoMigrate(&User{})
 	db.Model(&User{}).ModifyColumn("name", "varchar(30)")
 	db.Model(&User{}).ModifyColumn("token", "char(30)")
+	db.AutoMigrate(&Character{})
+	db.Model(&Character{}).ModifyColumn("name", "varchar(30)")
+	db.AutoMigrate(&UserCharacter{})
+	db.AutoMigrate(&GachaResult{})
 }
 
 func (u *User) CreateUser() *User {
-	rand.Seed(time.Now().UnixNano())
-
 	u.Token = generateRandomString(30)
 	db.NewRecord(u)
 	db.Create(&u)
 	return u
 }
 
 // Function defined to generate random string for token
-
 func generateRandomString(length int) string {
 	const charset = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789"
+	charsetLength := big.NewInt(int64(len(charset)))
 	b := make([]byte, length)
-	for i := range b {
-		b[i] = charset[rand.Intn(len(charset))]
+	for i := 0; i < length; i++ {
+		randomIndex, _ := rand.Int(rand.Reader, charsetLength)
+		b[i] = charset[randomIndex.Int64()]
 	}
 	return string(b)
 }
 
-func GetAllUsers() []User {
-	var Users []User
-	db.Find(&Users)
-	return Users
+func GetUserByToken(token string) []User {
+	var users []User
+	db.Where("token = ?", token).Find(&users)
```

Comment:
Was the token unique?
I think multiple users would be returned if there were duplicates.

Reply:
@iamananya Where is it guaranteed to be unique?

Reply:
@GuiltyMorishita  I have used a `generateRandomString` function so, its very unlikely that a token is ever repeated.